lib/Doctrine/DBAL/Driver/SQLAnywhere/SQLAnywhereConnection.php

```diff
@@ -137,13 +130,29 @@ public function getServerVersion()
     /**
      * {@inheritdoc}
      */
-    public function lastInsertId($name = null)
+    public function lastInsertId(?string $name = null) : string
     {
         if ($name === null) {
-            return sasql_insert_id($this->connection);
+            $result = sasql_insert_id($this->connection);
+
+            if ($result === false) {
+                throw SQLAnywhereException::fromSQLAnywhereError($this->connection);
+            }
+
+            if ($result === 0) {
+                throw new SQLAnywhereException('The last insert did not affect an AUTOINCREMENT column.');
```

Comment:
This message should be consistent across drivers. We can use a factory method like:

https://github.com/doctrine/dbal/blob/16ef12680c39f73cabffcf8814694ec2846ef54e/lib/Doctrine/DBAL/Schema/SchemaException.php#L24-L27